src/coreclr/src/jit/block.h

```diff
@@ -514,16 +514,17 @@ struct BasicBlock : private LIR::Range
     const char* dspToString(int blockNumPadding = 0);
 #endif // DEBUG
 
-    typedef unsigned weight_t; // Type used to hold block and edge weights
-                               // Note that for CLR v2.0 and earlier our
-                               // block weights were stored using unsigned shorts
-
-#define BB_UNITY_WEIGHT 100 // how much a normal execute once block weights
-#define BB_LOOP_WEIGHT 8    // how much more loops are weighted
-#define BB_ZERO_WEIGHT 0
-#define BB_MAX_WEIGHT UINT32_MAX // we're using an 'unsigned' for the weight
-#define BB_VERY_HOT_WEIGHT 256   // how many average hits a BB has (per BBT scenario run) for this block
-                                 // to be considered as very hot
+    // Type used to hold block and edge weights
+    typedef float weight_t;
+
+#define BB_UNITY_WEIGHT 100.0f       // how much a normal execute once block weights
+#define BB_UNITY_WEIGHT_UNSIGNED 100 // how much a normal execute once block weights
+#define BB_LOOP_WEIGHT 8.0f          // how much more loops are weighted
+#define BB_ZERO_WEIGHT 0.0f
+#define BB_HOT_WEIGHT 1000000.0f
+#define BB_MAX_WEIGHT FLT_MAX     // maximum finite weight  -- needs rethinking.
+#define BB_VERY_HOT_WEIGHT 256.0f // how many average hits a BB has (per BBT scenario run) for this block
```

Comment:
Why is BB_VERY_HOT_WEIGHT smaller than the new BB_HOT_WEIGHT?

Reply:
Good point.  I should rename these for what they are, they're each only used once.

BB_HOT_WEIGHT is used when we're prejtting and assessing the inlinability of the root method -- we assume there might be at least one call site with this weight.

BB_VERY_HOT_WEIGHT is used by the x86 emitter to decide if a method should be 16 byte aligned.

Reply:
Pushed an update.